src/libraries/System.Private.Xml/src/System/Xml/Core/XmlWriter.cs

```diff
@@ -490,7 +456,7 @@ public virtual void WriteNode(XPathNavigator navigator, bool defattr)
                             do
                             {
                                 IXmlSchemaInfo? schemaInfo = navigator.SchemaInfo;
-                                if (defattr || (schemaInfo == null || !schemaInfo.IsDefault))
+                                if (defattr || schemaInfo is not { IsDefault: true })
```

Comment:
I don't think pattern matching in a case like this is any more clear. Please revert.